File: src/342_power_of_four/PowerOfFour.java
public class PowerOfFour {
    public boolean isPowerOfFour(int n) {
        if (n <= 0)
            return false;
        while (n % 4 == 0) {
            n /= 4;
        }
        return n == 1;
    }

    public static void main(String[] args) {
        PowerOfFour sol = new PowerOfFour();
        System.out.println(sol.isPowerOfFour(16) + " | True");
        System.out.println(sol.isPowerOfFour(5) + " | False");
        System.out.println(sol.isPowerOfFour(1) + " | True");
        System.out.println(sol.isPowerOfFour(64) + " | True");
        System.out.println(sol.isPowerOfFour(0) + " | False");
        System.out.println(sol.isPowerOfFour(-4) + " | False");
    }
}

File: src/342_power_of_four/power_of_four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0:
            return False
        while n % 4 == 0:
            n //= 4
        return n == 1


def test():
    sol = Solution()
    print(sol.isPowerOfFour(16), "| True")
    print(sol.isPowerOfFour(5), "| False")
    print(sol.isPowerOfFour(1), "| True")
    print(sol.isPowerOfFour(64), "| True")
    print(sol.isPowerOfFour(0), "| False")
    print(sol.isPowerOfFour(-4), "| False")


if __name__ == "__main__":
    test()
